src/main/java/com/example/gymbuddy/infrastructure/entities/MachineHistory.java

```diff
@@ -0,0 +1,34 @@
+package com.example.gymbuddy.infrastructure.entities;
+
+import jakarta.persistence.*;
+import lombok.Data;
+
+import java.time.LocalDate;
+
+@Entity
+@Data
+public class MachineHistory {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @Column(name = "number_reps")
+    private Integer numberReps;
+
+    @Column(name = "max_weight")
+    private Integer maxWeight;
+
+    @Column(name = "number_sets")
+    private Integer numberSets;
+
+    @Column(name = "workout_date")
+    private LocalDate workoutDate;
+
+    @OneToOne
+    @JoinColumn(name = "member_id")
+    private Member member;
+
+    @OneToOne
+    @JoinColumn(name = "machine_id")
+    private Machine machine;
```

Comment:
Same with this, a machine will have many histories.

The one-to-one may work, but I'm not entirely sure how these annotations affect things

Reply:
Should it be @ManyToMany?